src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -4403,6 +4403,60 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/datasets/{id}/snapshotBuilder/count:
+    post:
+      tags:
+        - datasets
+        - repository
+      description: >
+        Given a list of snapshot builder cohorts, return the unique participant count for this dataset.
+      operationId: getSnapshotBuilderCount
+      parameters:
+        - in: path
+          description: The UUID of the dataset.
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        description: The cohorts to count participants in
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SnapshotBuilderCountRequest'
+        required: true
+      responses:
+        200:
+          description: The concepts with the specified concept as their parent
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SnapshotBuilderCountResponse'
+        401:
+          description: Unauthorized - dataset does not exist or missing permissions to view.
```

Comment:
Our other endpoints now return a 404 when a resource doesn't exist instead of a 401, so I think the snapshot builder endpoints should follow the same pattern for consistency